src/System.Private.CoreLib/shared/System/Buffers/ArrayPool.cs

```diff
@@ -19,6 +19,10 @@ namespace System.Buffers
     /// </remarks>
     public abstract class ArrayPool<T>
     {
+        // Store the shared ArrayPool in a field of its derived sealed type so the Jit can "see" the exact type
+        // when the Shared property is inlined which will allow it to devirtualize calls made on it.
+        private static TlsOverPerCoreLockedStacksArrayPool<T> s_shared = new TlsOverPerCoreLockedStacksArrayPool<T>();
```

Comment:
This should be also readonly. (The JIT does not take advantage of readonly for object references right now, but it can in future.)